types/react-transition-group/SwitchTransition.d.ts

```diff
@@ -0,0 +1,12 @@
+import { Component } from "react";
+import { TransitionProps } from "./Transition";
+
+declare namespace SwitchTransition {
+    interface SwitchTransitionProps extends TransitionProps {
+        mode?: 'out-in' | 'in-out';
```

Comment:
In the [original code](https://github.com/reactjs/react-transition-group/blob/f50fae0392e0beeb3989db9ddc59c04ecb20cf09/src/SwitchTransition.js#L23-L26) `modes` is also exported, so what about adding it like this? 👇 

```tsx
const modes = {
  out: 'out-in',
  in: 'in-out'
}

interface SwitchTransitionProps extends TransitionProps {
  mode?: modes.out | modes.in;
}
```

Reply:
@ybiquitous The above fails to build with an error "A 'const' initializer in an ambient context must be a string or numeric literal or literal enum reference"

Reply:
Oh, sorry. It seems good to use `enum` instead of `const` here.

By the way, I notice PR #37061 has been opened. 👀

Reply:
@ybiquitous thanks, PR #37061 does what I wanted it to :)